WcaOnRails/app/controllers/registrations_controller.rb

```diff
@@ -195,7 +200,7 @@ def register_require_sign_in
   def register
     @competition = competition_from_params
     @registration = nil
-    @selected_events ||= []
+    @selected_events = []
     if current_user
       @registration = Registration.find_by(user_id: current_user.id, competition_id: @competition.id) || @competition.registrations.build(user_id: current_user.id)
       @selected_events = @registration.saved_and_unsaved_events.empty? ? @registration.user.preferred_events : @registration.saved_and_unsaved_events
```

Comment:
1. I did not know about `find_or_initialize_by`, that looks like an improvement!
2. Using `saved_and_unsaved_events` vs `events`. I don't have an opinion on this. I was inclined to leave it as `saved_and_unsaved_events` everywhere, just so future code readers don't have to think through "why do they sometimes call `saved_and_unsaved_events` and sometimes just call `events`?" If you feel strongly about calling `events` here, then I'll make the switch.